generator/src/main/java/org/apache/kafka/message/ApiMessageTypeGenerator.java

```diff
@@ -245,6 +247,58 @@ private void generateToString() {
         buffer.printf("}%n");
     }
 
+    private void generateHeaderVersion() {
+        buffer.printf("public short headerVersion(short version) {%n");
```

Comment:
I wonder if it would make sense to create two different methods here to respectively retrieve the version of the request header and the version of the response header. I think we will need this to support the ApiVersions case. We could talk this in a separate PR or as part of KIP-511. What do you think?

Reply:
yes, let's do that.